From 09379f1bf2af662da167f5efa05915960864973d Mon Sep 17 00:00:00 2001
From: Kever Yang <kever.yang@rock-chips.com>
Date: Tue, 5 Mar 2019 10:22:19 +0800
Subject: [PATCH] ARM: dts: rockchip: add battery node for rk3128-fireprime

fireprime is using rk818, which need a battery node for fuel gauge,
or else the input current will be limit to 500mA.

Change-Id: Ie80dbc103d1ac57b704235a9b618b7e9db44c953
Signed-off-by: Kever Yang <kever.yang@rock-chips.com>
---
 arch/arm/boot/dts/rk3128-fireprime.dts | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm/boot/dts/rk3128-fireprime.dts b/arch/arm/boot/dts/rk3128-fireprime.dts
index a75d814f91a7..8cf91aa5d53b 100644
--- a/arch/arm/boot/dts/rk3128-fireprime.dts
+++ b/arch/arm/boot/dts/rk3128-fireprime.dts
@@ -341,6 +341,30 @@
 				};
 			};
 		};
+
+		battery {
+			compatible = "rk818-battery";
+			ocv_table = <3400 3650 3693 3707 3731 3749 3760
+				     3770 3782 3796 3812 3829 3852 3882
+				     3915 3951 3981 4047 4086 4132 4182>;
+			design_capacity = <4000>;
+			design_qmax = <4000>;
+			bat_res = <100>;
+			max_input_current = <2000>;
+			max_chrg_current = <1400>;
+			max_chrg_voltage = <4200>;
+			sleep_enter_current = <300>;
+			sleep_exit_current = <300>;
+			power_off_thresd = <3400>;
+			zero_algorithm_vol = <3850>;
+			energy_mode = <0>;
+			fb_temperature = <105>;
+			sample_res = <20>;
+			max_soc_offset = <60>;
+			monitor_sec = <5>;
+			virtual_power = <1>;
+			power_dc2otg = <0>;
+		};
 	};
 };
 
-- 
2.35.3

